src/electionguard_cli/cli_steps/output_step_base.py

```diff
@@ -0,0 +1,47 @@
+from typing import List, Any
+from os.path import join
+
+from electionguard import to_file
+from electionguard.guardian import Guardian, GuardianRecord
+from electionguard_tools.helpers.export import GUARDIAN_PREFIX
+
+from .cli_step_base import CliStepBase
+from ..cli_models import CliElectionInputsBase
+
+
+class OutputStepBase(CliStepBase):
+    """Responsible for common functionality across all CLI commands related to outputting results."""
+
+    def _export_private_keys(self, output_keys: str, guardians: List[Guardian]) -> None:
+        if output_keys is None:
+            return
+
+        private_guardian_records = [
+            guardian.export_private_data() for guardian in guardians
+        ]
+        file_path = output_keys
+        for private_guardian_record in private_guardian_records:
+            file_name = GUARDIAN_PREFIX + private_guardian_record.guardian_id
+            to_file(private_guardian_record, file_name, file_path)
+        self.print_value("Guardian private keys", output_keys)
+        self.print_warning(
+            f"The files in {file_path} are secret and should be protected securely and not shared publicly."
+        )
+
+    @staticmethod
+    def _get_guardian_records(
```

Comment:
This is only used publicly but is named privately

Reply:
Also this seems like a method to just put on the inputs base?

Reply:
I really don't know what to do with this, and it's a pattern I'm struggling with constantly in Python.  In most OO languages it'd be a protected virtual instance method that the inheriting classes would call on their base class.  The inheriting classes might even choose to override some of that functionality if they wanted.  So that's why I don't consider this usage scenario to be public, because the two inheriting classes `ElectionRecordStep` and `OutputSetupFilesStep` are just calling it on their parent class `OutputStepBase`.

However, if I make `_get_guardian_records` an instance method then the linter complains `Method could be a function (no-self-use)`.  But if I solve that by making it static then the base classes can't override it, and worse it's suddenly considered public in Python?  And once it's static then it's considered a code smell.  

Any guidance would be greatly appreciated, keeping in mind this is just one instance of a pattern I'm running into frequently, so the consequences of this conversation reach much farther than this one instance.

Reply:
This doesn't reference anything in this class. I would put this on election_inputs since that is actually what is using this and simply add it as a getter. 

```py
class ElectionInputs()
        ...
        def get_election_records(self) -> List[GuardianRecord]:
                return [guardian.publish() for guardian in self.guardians]
```

Reply:
Otherwise this code can be just left as is.

Reply:
True, although ElectionInputs is currently a dataclass.  Doesn't the existing architecture favor anemic domain models?  I don't mind going against that, I'm just trying to understand when to use that pattern and when not to, and I'm especially interested because this problem seems to keep cropping up for me.